rust/agama-dbus-server/src/network/nm/dbus.rs

```diff
@@ -57,25 +59,25 @@ pub fn connection_to_dbus(conn: &Connection) -> NestedHash {
 ///
 /// This functions tries to turn a OwnedHashMap coming from D-Bus into a Connection.
 pub fn connection_from_dbus(conn: OwnedNestedHash) -> Option<Connection> {
-    let base = base_connection_from_dbus(&conn)?;
+    let mut connection = base_connection_from_dbus(&conn)?;
```

Comment:
Not sure if it is a problem at all. This function show the case that we now easily transform a Connection from one type into the other, by setting the `config` enum. Before it was like, collecting common settings + specific settings and then construct the Connection. 

But currently I do not see a problem with it. For me this approach has benefits over the current.

Reply:
Yes, I have the same opinion and it does not look like a problem (at least by now). I think this approach is simpler. But I will work on this once the bonding support is finally merged.